File: leetcode/strings/344.go
package main

import "fmt"

func reverseString(s []byte) {

	var i int = 0
	var j int = len(s) - 1

	for {
		s[i], s[j] = s[j], s[i]

		i++
		j--

		if i >= j {
			break
		}
	}
}

func main() {
	str := []byte{'h', 'e', 'l', 'l', 'o'}
	reverseString(str)
	fmt.Println(str)
}

File: leetcode/data_structures/array/344.py
from typing import List


def reverseString(s: List[str]) -> None:

    i = 0
    j = len(s)-1

    while i < j:
        a[i], a[j] = a[j], a[i]
        i += 1
        j -= 1


a = ["h", "e", "l", "l", "o"]
reverseString(a)
print(a)
